doc/access.md

```diff
@@ -1,49 +1,162 @@
-give access to any of the secrets within. For more info see the repo's README.
-
-### Test machines
-
-[@nodejs/build][] have root access to the test CI machines (`test-*`). The list
-of members is [here][Build WG Members].
-
-### Infra machines
-
-A subsection of build members have access to infra machines
-(`infra-*`). The list of members is [here][Infra Admins].
-
-The infra group also have access to:
-
-#### Servers:
+* [Build Working Group Membership](#build-working-group-membership)
+* [Resource Access](#resource-access)
+  * [Test servers](#test-servers)
+  * [Infra servers](#infra-servers)
+    * [IaaS services](#iaas-services)
+    * [Other services](#other-services)
+    * [Certificates](#certificates)
+  * [Release servers](#release-servers)
+    * [Certificates](#certificates-1)
+  * [ci.nodejs.org](#cinodejsorg)
+    * [Jenkins admins](#jenkins-admins)
+  * [ci-release.nodejs.org](#ci-releasenodejsorg)
+  * [[GitHub Bot][]](#github-bot)
+* [NPM Management](#npm-management)
+
+This document describes which groups have access to which assets managed by the
+Build Working Group and how membership of those groups is managed.
+
+_Note that links to `@nodejs/` teams in this document are not visible to people
+who aren't in the Nodejs organization, and the [secrets repo][] is only visible
+to people who have read access to the nodejs-private organization._
+
+For technical details on getting SSH access to the machines, see the
+[SSH guide][].
+
+## Build Working Group Membership
+
+The Build WG is comprised of individuals who are interested in managing servers
+and the services that are managed on behalf of the Node.js project. Membership
+is determined by the group itself, with existing, active, members voting on
+proposals to add new members. You can see existing [Build WG members][] on the
+README.
+
+When considering new members, the Build WG is primarily concerned with
+**competence** and **trust**. Membership grants access to a significant amount
+of, resources. While we partition our resources into trust levels, the basic
+level that all members have access to comprises the largest number of servers
+maintained by the Build WG. Members should have a basic level of competence in
+one or more technical areas covered by the Build WG such that the addition of
+a new member spreads the maintenance burden rather than creating additional
+burden because of lack of competence, or un-trustworthy behavior. The Build WG
+is not an expert group and we offer a place to learn and develop skills. Members
+should be aware of the bounds of their expertise and act accordingly.
+
+* Competence: it is difficult to objectively gauge technical competence without
+  demonstration. You can demonstrate competence by contributing to resources in
+  this repository where you see need or attempting to assist Build WG members
+  where you are able. Competence may also be demonstrated through contributions
```

Comment:
I'm very, very sympathetic to the risks of incautious access, but I worry that this veers close to a catch-22. Despite that sense that we have an intimidating mountain of work available, I'm struggling to find helpful things that can be done without access, and without doing helpful things, its hard to demonstrate competence. @richardlau and I both demonstrated competence the long way: nodejs collaboration.

Reply:
There are a few things that can be contributed without access, for example I have [four pull requests](https://github.com/nodejs/build/pulls?q=is%3Apr+author%3Arichardlau+is%3Aclosed) merged into this repository that happened before I was given access ("contributing to resources in this repository where you see need"). Since collaborators can view (but not change) job configuration in Jenkins it's even possible to [suggest fixes](https://github.com/nodejs/node/issues/24349#issuecomment-440490499) to jobs ("attempting to assist Build WG members where you are able").

I wouldn't necessarily say it's easy -- you need some idea of what is where.

Reply:
Agree, this is a very difficult corner but this PR is intended to clarify current thinking and open discussion for amending that thinking where we can. So if you have suggestions for how to change things then speak up!